src/java.base/macosx/classes/sun/nio/ch/KQueuePoller.java

```diff
@@ -34,13 +38,13 @@ class KQueuePoller extends Poller {
     private final int kqfd;
     private final int filter;
     private final int maxEvents;
-    private final long address;
+    private final MemorySegment pollArray;
 
     KQueuePoller(boolean subPoller, boolean read) throws IOException {
-        this.kqfd = KQueue.create();
+        this.kqfd = kqueue_h.kqueue();
```

Comment:
Note that the old implementation of `KQueue.create()` had a check for negative `kqfd`:
https://github.com/openjdk/jdk/blob/d288ca28be7bfba3abe9f54cefbe53e73c25707e/src/java.base/macosx/native/libnio/ch/KQueue.c#L63-L71